dotnet/src/InternalUtilities/src/Diagnostics/Verify.cs

```diff
@@ -84,7 +84,7 @@ internal static void DirectoryExists(string path)
     /// Make sure every function parameter name is unique
     /// </summary>
     /// <param name="parameters">List of parameters</param>
-    internal static void ParametersUniqueness(IList<ParameterView> parameters)
+    internal static void ParametersUniqueness(IReadOnlyList<ParameterView> parameters)
     {
         int count = parameters.Count;
```

Comment:
I don't see a need to use index in here + This method is specific to ParameterView, might worth moving it as an internal impl of it used in the `ctor`

Reply:
Gotcha. Room for improvement in this function. I've filed #2908 to clean up this function and where it's implemented/called.